crates/sui-types/src/messages.rs

```diff
@@ -1916,6 +1916,11 @@ pub enum ExecutionFailureStatus {
     /// In PaySui/PayAllSui, it means some input coins are not SUI coins.
     CoinTypeMismatch,
     CoinTooLarge,
+    /// The total amount of coins to be paid is larger than the maximum value of u64
+    TotalAmountOverflow,
+    /// The total amount of input coins is larger than the maximum value of u64,
+    /// which should never happen.
```

Comment:
should never happen b/c of the SUI total token cap.

Reply:
I don't think we want to add error types for outcomes that are not possible--should be a `panic` or generic `InvariantViolation` instead (@tnowacki can advise on error discipline)